File: app/src/main/java/com/example/newdiary/MyToast.java
package com.example.newdiary;

import android.content.Context;
import android.widget.Toast;

public class MyToast {
    public static Toast mToast;
    public static void showMsg(Context context, String msg){
        if(mToast == null){
            mToast = Toast.makeText(context,msg,Toast.LENGTH_LONG);
        }else {
            mToast.setText(msg);
        }
        mToast.show();
    }

}

File: app/src/main/java/com/example/newdiary/MyToast.kt
package com.example.newdiary

import android.content.Context
import android.widget.Toast

class MyToast {
    companion object {
        private var mToast: Toast? = null

        // todo 代码可以精简
        fun showToast(context: Context, msg: String, isLong: Boolean) {
            mToast = if (mToast == null) {
                if (isLong) {
                    Toast.makeText(context, msg, Toast.LENGTH_LONG)
                } else {
                    Toast.makeText(context, msg, Toast.LENGTH_SHORT)
                }
            } else {
                mToast?.cancel()
                if (isLong) {
                    Toast.makeText(context, msg, Toast.LENGTH_LONG)
                } else {
                    Toast.makeText(context, msg, Toast.LENGTH_SHORT)
                }
            }
            mToast?.show()
        }
    }

}